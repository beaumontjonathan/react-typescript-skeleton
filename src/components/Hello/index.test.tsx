import { shallow } from 'enzyme';
import * as React from 'react';
import { Hello } from './index';

describe('(component) Hello', () => {
  it('should render without crashing', () => {
    const helloComponent = shallow(<Hello framework="react" compiler="webpack" />);

    expect(helloComponent.type()).toBe('h1');
  });

  it('should be styled correctly', () => {
    const helloComponent = shallow(<Hello framework="react" compiler="webpack" />);

    expect(helloComponent).toMatchSnapshot();
  });

  it('should take and apply a framework prop', () => {
    const helloComponent = shallow(<Hello framework="react" compiler="webpack" />);

    expect(helloComponent.text().includes('react')).toBeTruthy();
  });

  it('should take and apply a compiler prop', () => {
    const helloComponent = shallow(<Hello framework="react" compiler="webpack" />);

    expect(helloComponent.text().includes('webpack')).toBeTruthy();
  });
});
