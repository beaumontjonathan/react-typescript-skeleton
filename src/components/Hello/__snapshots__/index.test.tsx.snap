// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(component) Hello should be styled correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    compiler="webpack"
    framework="react"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Hello from ",
        "webpack",
        " and ",
        "react",
        "!",
      ],
    },
    "ref": null,
    "rendered": Array [
      "Hello from ",
      "webpack",
      " and ",
      "react",
      "!",
    ],
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Hello from ",
          "webpack",
          " and ",
          "react",
          "!",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Hello from ",
        "webpack",
        " and ",
        "react",
        "!",
      ],
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
